src/schemathesis/models.py

```diff
@@ -216,14 +221,25 @@ def as_werkzeug_kwargs(self, headers: Optional[Dict[str, str]] = None) -> Dict[s
             **extra,
         }
 
-    def call_wsgi(self, app: Any = None, headers: Optional[Dict[str, str]] = None, **kwargs: Any) -> WSGIResponse:
+    def call_wsgi(
+        self,
+        app: Any = None,
+        headers: Optional[Dict[str, str]] = None,
+        **kwargs: Any
+    ) -> WSGIResponse:
         application = app or self.app
         if application is None:
             raise RuntimeError(
                 "WSGI application instance is required. "
                 "Please, set `app` argument in the schema constructor or pass it to `call_wsgi`"
             )
         data = self.as_werkzeug_kwargs(headers)
+        if "headers" not in data or data["headers"] is None:
+            data.update({"headers": {}})
+        if headers is not None:
+            data["headers"].update(headers)
+        if "User-Agent" not in data["headers"]:
+            data["headers"].update({"User-Agent": USER_AGENT})
```

Comment:
Instead of using `update` with a single key, I'd prefer using `__setitem__` - `data["headers"]["User-Agent"] = USER_AGENT`. It is more readable and slightly faster